sql3/test/defs/defs_string_functions.go

```diff
@@ -820,5 +820,87 @@ var stringScalarFunctionsTests = TableTest{
 			),
 			ExpErr: "[0:0] value '-1' out of range",
 		},
+		{
+
+			name: "CharIndexIncorrectNumberofArguments",
+			SQLs: sqls(
+				"select charindex('is','this is great',3,4)",
+			),
+			ExpErr: "'charindex': count of formal parameters (3) does not match count of actual parameters (4)",
+		},
+		{
+			name: "CharIndexIncorrectTypeofArgumentsforPosition",
+			SQLs: sqls(
+				"select charindex('is','this is great','you')",
+			),
+			ExpErr: "integer expression expected",
+		},
+		{
+			name: "CharIndexIncorrectTypeofArgumentsforInputString",
+			SQLs: sqls(
+				"select charindex('is',23,3)",
+			),
+			ExpErr: "string expression expected",
```

Comment:
add compareexactordered

Reply:
Do we need to add compareexactordered when we are expecting errors?

Reply:
no, you don't need it for errors